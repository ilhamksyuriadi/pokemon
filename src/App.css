html,body{
  padding: 0;
  margin: 0
}

.App {
  text-align: center;
  font-family: 'Sigmar One', cursive;
}

.page-container{
  min-height: 400px;
}
