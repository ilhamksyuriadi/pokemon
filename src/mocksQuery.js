// query for testing
import { GET_POKEMON_DETAIL } from './operations/queries/getPokemon';
import { GET_POKEMON_LIST } from './operations/queries/getPokemonList';

export const mocksPokemonDetail = [
    {
        request: {
            query: GET_POKEMON_DETAIL,
            variables: {
                name: 'ditto',
            },
        },
        result: {
            data: {
                pokemon: {
                    id: 132,
                    name: "ditto",
                    abilities: [
                        {
                            ability: {
                                name: "limber"
                            }
                        },
                        {
                            ability: {
                                name: "imposter"
                            }
                        }
                    ],
                    moves: [
                        {
                            move: {
                                name: "transform"
                            }
                        }
                    ],
                    types: [
                        {
                            type: {
                                name: "normal"
                            }
                        }
                    ]
                }
            },
        },
    },
];

export const mocksPokemonList = [
    {
        request: {
            query: GET_POKEMON_LIST,
            variables: {
                limit: 2,
                offset: 0
            },
        },
        result: {
            data: {
                pokemons: {
                    results: [
                        {
                            name: "bulbasaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                        },
                        {
                            name: "ivysaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
                        },
                        {
                            name: "venusaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
                        }
                    ]
                }
            }
        },
    },
];

export const mocksApp = [
    {
        request: {
            query: GET_POKEMON_DETAIL,
            variables: {
                name: 'ditto',
            },
        },
        result: {
            data: {
                pokemon: {
                    id: 132,
                    name: "ditto",
                    abilities: [
                        {
                            ability: {
                                name: "limber"
                            }
                        },
                        {
                            ability: {
                                name: "imposter"
                            }
                        }
                    ],
                    moves: [
                        {
                            move: {
                                name: "transform"
                            }
                        }
                    ],
                    types: [
                        {
                            type: {
                                name: "normal"
                            }
                        }
                    ]
                }
            },
        },
    },
    {
        request: {
            query: GET_POKEMON_LIST,
            variables: {
                limit: 2,
                offset: 0
            },
        },
        result: {
            data: {
                pokemons: {
                    results: [
                        {
                            name: "bulbasaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                        },
                        {
                            name: "ivysaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
                        },
                        {
                            name: "venusaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
                        }
                    ]
                }
            }
        },
    },
]