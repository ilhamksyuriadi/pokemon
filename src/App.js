import './App.css';
import PokemonList from './views/PokemonList';
import MyPokemon from './views/MyPokemon';
import PokemonDetail from './views/PokemonDetail';
import Header from './components/Header';
import Footer from './components/Footer';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";

function App() {
  return (
    <Router>
      <div className="App">
        <Header />
        <Switch>
          <Route exact path="/mypokemon">
            <MyPokemon />
          </Route>
          <Route exact path="/pokemon/:name">
            <PokemonDetail />
          </Route>
          <Route exact path="/">
            <PokemonList />
          </Route>
        </Switch>
        <Footer />
      </div>
    </Router>
  );
}

export default App;
