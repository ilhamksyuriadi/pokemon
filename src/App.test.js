import App from './App';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";
import { MockedProvider } from '@apollo/client/testing';
import { mocksApp } from './mocksQuery';

const mocks = mocksApp;
let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("full app, snapshot", () => {
    act(() => {
        render(
            <MockedProvider mock={mocks}>
                <App />
            </MockedProvider>
        ,container)
    })
    expect(container).toMatchSnapshot();
})