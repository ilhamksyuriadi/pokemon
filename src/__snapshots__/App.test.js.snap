// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full app, snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header>
      <h1>
        Pokemon App
      </h1>
      <nav>
        <a
          aria-current="page"
          class="nav-item nav-active"
          href="/"
        >
          <h3>
            Pokemon List
          </h3>
        </a>
        <a
          class="nav-item"
          href="/mypokemon"
        >
          <h3>
            My Pokemon
          </h3>
        </a>
      </nav>
    </header>
    <div
      class="loading-layout"
    >
      <div>
        <img
          alt="pidgey"
          src="pidgey-front.png"
        />
        <h4>
          getting pokemon list...
        </h4>
      </div>
    </div>
    <footer>
      <div>
        <img
          alt="logo"
          class="game-freak-logo"
          src="game-freak.png"
        />
        <p>
          ilhamksyuriadi Â©2020
        </p>
      </div>
    </footer>
  </div>
</div>
`;
